src/status_im/ui/screens/wallet/send/views/common.cljs

```diff
@@ -0,0 +1,453 @@
+            [status-im.i18n :as i18n]
+            [clojure.string :as string]
+            [status-im.ui.components.tooltip.views :as tooltip]
+            [status-im.ui.screens.wallet.send.events :as events]
+            [status-im.utils.ethereum.core :as ethereum]
+            [status-im.utils.dimensions :as dimensions]
+            [taoensso.timbre :as log]
+            [status-im.utils.ethereum.tokens :as tokens]))
+
+(defn toolbar [flow title chat-id]
+  (let [action (if (#{:chat :dapp} flow) actions/close-white actions/back-white)]
+    [toolbar/toolbar {:style wallet.styles/toolbar}
+     [toolbar/nav-button (action (if (= :chat flow)
+                                   #(re-frame/dispatch [:chat.ui/navigate-to-chat chat-id {}])
+                                   #(actions/default-handler)))]
+     [toolbar/content-title {:color :white :font-size 17} title]]))
+
+(defn action-button [{:keys [disabled? on-press underlay-color background-color]} content]
+  [react/touchable-highlight {:underlay-color underlay-color
+                              :disabled       disabled?
+                              :on-press       on-press
+                              :style          {:height           44
+                                               :background-color background-color
+                                               :border-radius    8
+                                               :flex             1
+                                               :align-items      :center
+                                               :justify-content  :center
+                                               :margin           3}}
+   content])
+
+(defn info-page [message]
+  [react/view {:style {:flex             1
+                       :align-items      :center
+                       :justify-content  :center
+                       :background-color colors/blue}}
+   [vector-icons/icon :main-icons/info {:color colors/white}]
+   [react/text {:style {:max-width   144
+                        :margin-top  15
+                        :color       colors/white
+                        :font-size   15
+                        :text-align  :center
+                        :line-height 22}}
+    message]])
+
+;; worthy abstraction
+(defn anim-ref-send
+  "Call one of the methods in an animation ref.
+   Takes an animation ref (a map of keys to animation tiggering methods)
+   a keyword that should equal one of the keys in the map  and optional args to be sent to the animation.
+
+   Example:
+    (anim-ref-send slider-ref :open!)
+    (anim-ref-send slider-ref :move-top-left! 25 25)"
+  [anim-ref signal & args]
+  (when anim-ref
+    (assert (get anim-ref signal)
+            (str "Key " signal " was not found in animation ref. Should be in "
+                 (pr-str (keys anim-ref)))))
+  (some-> anim-ref (get signal) (apply args)))
+
+(defn slide-up-modal
+  "Creates a modal that slides up from the bottom of the screen and
```

Comment:
I think we  have a bottom sheet component or something (for instance when we create a new chat), can it be reused here? Or this one is different?

Reply:
Yeah, I know. This code had originally been written before that component was introduced. The consensus we got on this is to merge it as it is and then refactor-replace this occurence with the new component. The reason is to reduce the scope creep of the pr.

Reply:
understood